rfcs/text/0000-image-base-transcriptomics.md

```diff
@@ -0,0 +1,601 @@
+aligned, background removed, and spots computationally enhanced.
+*starfish* can output these processed images, annotated with information
+sufficient to re-do the processing, so that the images input to spot
+calling can be evaluated by users. If emitted, processed images will be
+generated in SpaceTx Format.
+
+#### HCA Sample Metadata
+
+The HCA metadata team has built a metadata specification draft for
+imaging data that references data in SpaceTx-Format. The issue that
+formed the draft can be found
+[<span class="underline">here</span>](https://github.com/HumanCellAtlas/metadata-schema/issues/368).
+
+### Data and Metadata validation
+
+#### Input Data validation
+
+SpaceTx Format consists of a json file that specifies how to compose a
+dataset from a set of two-dimensional images. As a result, it is
+possible to rapidly validate that a file is complete and internally
+consistent. In addition, SpaceTx Format stores per-chunk checksum
+information, enabling the data to be checked for integrity. It is
+possible to carry out additional, more data-specific checks, such as
+that images are not blank, contain spots, or contain cells, however
+these checks take increasing amounts of time.
+
+OME and SpaceTx can provide a docker container that validates image
+datasets with the highest level of stringency possible in a DCP-provided
+"maximum validation time".
+
+#### Output Data validation
+
+Each of the output formats produced by SpaceTx can be stored in Zarr and
+read (on-demand) by *starfish* and xarray. SpaceTx provides a docker
+container that loads and validates each output file using the xarray
+library. Validation can be tuned to occur within a DCP-provided target
+validation time.
+
+### Data Processing Workflows
+
+To process data in the DCP, the *starfish* team can deliver WDL-wrapped
+data processing workflows for any of the existing image-based
+transcriptomics pipelines. These workflows are based on the *starfish*
+python library, which in turn leverages standard open-source python
+projects. *starfish* provides a docker container that exposes both a
+python API and a CLI, and supports composable, modular, pipelines. These
+pipelines take as inputs images in SpaceTx Format and a pipeline recipe.
+Each assay emits output in the formats specified above. More detail on
+the state of image-based transcriptomics analyses can be found
+[<span class="underline">here</span>](https://docs.google.com/document/d/18Ve9C_cEn5lI2paivfgTAyjsM62qMyUF_s4mj82l1L0/edit#).
+
+### Data Processing Infrastructure
+
+All *starfish* pipelines share the same inputs and outputs. It should
+therefore be possible to deliver a single adapter WDL that satisfies any
+SpaceTx assay, and a family of subscriptions with minimal inter-class
+changes that specify the particular assay in question.
+
+### Delivery of Output Data to Users
+
+#### Matrices
+
+As mentioned above, image-based transcriptomics output data adhere to
```

Comment:
This may not be true anymore. @mckinsel is best positioned to provide additional details, but the matrix service new implementation no longer reads in zarr stores from dss bundles.

Reply:
Makes sense. @joshmoore I think it's probably sufficient to replace this paragraph with:

```
As mentioned above, image-based transcriptomics output data 
include a gene expression matrix and related metadata that are 
compatible with matrix service requirements and should be no
more difficult to convert than the output of a sequencing assay
```